tacs/problems/static.py

```diff
@@ -752,86 +903,169 @@ def solve(self, Fext=None):
 
         initSolveTime = time.time()
 
-        # Get current residual
-        self.getResidual(self.res, Fext=Fext)
+        if self.isNonlinear:
+            self.solveNonlinear(Fext)
+        else:
+            # Get current residual
+            self.getResidual(self.res, Fext=Fext)
 
-        # Get rhs vector
-        self.K.mult(self.u, self.rhs)
-        self.rhs.axpy(-1.0, self.res)
+            # Get rhs vector
+            self.K.mult(self.u, self.rhs)
+            self.rhs.axpy(-1.0, self.res)
 
-        # Set initnorm as the norm of rhs
-        self.initNorm = np.real(self.rhs.norm())
+            # Set initnorm as the norm of rhs
+            self.initNorm = np.real(self.rhs.norm())
 
-        # Starting Norm for this computation
-        self.startNorm = np.real(self.res.norm())
+            # Starting Norm for this computation
+            self.startNorm = np.real(self.res.norm())
 
-        initNormTime = time.time()
+            initNormTime = time.time()
 
-        # Solve Linear System for the update
-        success = self.KSM.solve(self.res, self.update)
+            # Solve Linear System for the update
+            self.solveJacLinear(self.res, self.update)
```

Comment:
Good idea, done

Reply:
Is there a reason why `solveNonLinear` and `solveLinear` need to be public methods? Will a user ever need to call the methods directly? Otherwise, we should make them private (i.e. `_solveNonLinear` and `_solveLinear`)

Reply:
not sure about `solveNonlinear` but a `solveLinear` public method definitely seems useful in the same way that we currently have a public `solveAdjoint` method. It would allow to support forward mode in the MPhys interface.

Reply:
Unless, there's a specific place where we can use these as public methods, lets make them private (we can always change this later). TACS doesn't support forward mode derivatives anyways, so I'm still not convinced its needed